src/lib.rs

```diff
@@ -163,53 +172,175 @@ pub mod pallet {
+
+            NonceCall::<T>::insert(sindex, nonce + 1);
+			Ok(())
 		}
 
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn approve_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			call_index: CallIndex,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+
+			if !Self::is_user_in_supersig(sindex, &who) {
+				return Err(Error::<T>::NotMember.into())
+			}
+            if Self::users_votes((sindex, call_index), who.clone()) {
+                return Err(Error::<T>::AlreadyVoted.into())
+            }
+            if Self::calls(sindex, call_index).is_none() {
+                return Err(Error::<T>::CallNotFound.into())
+            }
+
+            UsersVotes::<T>::insert((sindex, call_index), who.clone(), true);
+            Votes::<T>::insert(sindex, call_index, Self::votes(sindex, call_index) + 1);
+
+            Self::deposit_event(Event::<T>::CallVoted(supersig_id, call_index, who));
+
+            let threshold = Self::supersigs(sindex).unwrap().threshold;
+            let total_votes = Self::votes(sindex, call_index);
+
+            if total_votes >= threshold {
+                Self::execute_call(sindex, call_index);
+            }
+
+            Ok(())
+        }
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn remove_call(
+			origin: OriginFor<T>,
+			supersig_id: T::AccountId,
+			call_index: CallIndex,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let sindex = Self::get_supersig_index_from_id(&supersig_id).ok_or(Error::<T>::SupersigNotFound)?;
+            if who != supersig_id {
+                return Err(Error::<T>::NotAllowed.into());
+            }
+            if Self::calls(sindex, call_index).is_none() {
+                return Err(Error::<T>::CallNotFound.into())
+            }
+            Self::unchecked_remove_call(sindex, call_index);
+            Ok(())
+        }
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn get_supersig_index_from_id(id: &T::AccountId) -> Option<u128> {
+			let res = PalletId::try_from_sub_account(id).map(|(_, index)| index);
+            if let Some(index) = res {
+                if index < Self::nonce_supersig() {
+                    return Some(index)
+                } else {
+                    return None
+                }
+            } else {
+                None
+            }
+		}
+		pub fn get_supersig_id_from_index(index: u128) -> T::AccountId {
+            T::PalletId::get().into_sub_account(index)
+		}
 		pub fn is_user_in_supersig(supersig_id: u128, user: &T::AccountId) -> bool {
 			match Self::supersigs(supersig_id).map(|supersig| supersig.members.contains(user)) {
 				None => false,
 				Some(r) => r,
 			}
 		}
+
+        pub fn execute_call(supersig_index: u128, call_index: u128) {
+            let preimage = Self::calls(supersig_index, call_index).unwrap();
```

Comment:
No unwrap without comment explaining why it safe.
It safe here because it's only called at one place, in `approve_call` where you know those indexes exist.

But maybe later somebody will call it in another context.
So in this case I wouldn't even rely on simple comment.

The best option there is to inline the function.
Put it back in the `if threshold met`

Reply:
changed the `let ... = ....unwrap()` to a `if let ...` to avoid the unwrap, and keep it in helper function